server/api/main.py

```diff
@@ -747,25 +747,34 @@ async def _stop_logs():
         await fastapi.concurrency.run_in_threadpool(close_session, db_session)
 
 
-async def _stop_logs_for_runs(runs: list):
-    project_to_run_uids = {}
+async def _stop_logs_for_runs(runs: list, chunk_size: int = 10):
```

Comment:
this method is not being run in threadpool. I believe it blocks the main thread like we talked.

Reply:
there is nothing to run here in a thread. the code within is fully asyncio compatible (awaitable)